src/Common/HashTable/Hash.h

```diff
@@ -48,15 +48,22 @@ inline DB::UInt64 intHash64(DB::UInt64 x)
 #include <arm_acle.h>
 #endif
 
+#include <zlib.h>
+
+inline uint64_t zlibCRC32(uint64_t crc, uint64_t data)
+{
+    return crc32(crc, reinterpret_cast<const unsigned char*>(&data), sizeof(data));
+}
+
 inline DB::UInt64 intHashCRC32(DB::UInt64 x)
 {
 #ifdef __SSE4_2__
     return _mm_crc32_u64(-1ULL, x);
 #elif defined(__aarch64__) && defined(__ARM_FEATURE_CRC32)
     return __crc32cd(-1U, x);
 #else
-    /// On other platforms we do not have CRC32. NOTE This can be confusing.
-    return intHash64(x);
+    /// On other platforms we use zlib CRC
+    return zlibCRC32(-1ULL, x);
```

Comment:
I don't think this is a good option.

CRC-32 has lower quality than intHash64,
and it has lower performance on hardware platforms without the corresponding instructions.

Reply:
@alexey-milovidov, I rollbacked the change of Hash.h and updates thresholds in weakhash32 unit test for s390x, please review again. Thanks.